python/datasources/acs_health_insurance.py

```diff
@@ -0,0 +1,612 @@
+
+        # Split internal memory into data frames for sex/race by state/county
+        self.split_data_frames()
+
+        # Create BQ columns and write dataframes to BQ
+        for table_name, df in self.frames.items():
+            # All breakdown columns are strings
+            column_types = {c: 'STRING' for c in df.columns}
+
+            column_types[WITH_HEALTH_INSURANCE_COL] = 'INT64'
+            column_types[WITHOUT_HEALTH_INSURANCE_COL] = 'INT64'
+            column_types[TOTAL_HEALTH_INSURANCE_COL] = 'INT64'
+
+            gcs_to_bq_util.append_dataframe_to_bq(
+                df, dataset, table_name, column_types=column_types)
+
+    # This method runs through each Race/Sex Prefix Suffix
+    # group and creates an in memory metadata table
+    # by combining the prefix metadata with the suffix metadata. This makes lookup of the combined
+    # Metadata quick, without having to explicitely define each combination of metadata
+    # EG:
+    # Prefixes:
+    # {"C27001A":{MetadataKey.RACE: RACE["WHITE_ALONE"]}},
+    # {"C27001B":{MetadataKey.RACE: RACE["BLACK_OR_AFRICAN_AMERICAN_ALONE"]}}
+    # Suffixes:
+    #    {"_002E":meta(None,0, 19),
+    #     "_003E":meta(None,0, 19,HealthInsurancePopulation.WITH) ``}
+    # Combine Into Four:
+    # {"C27001A_002E": {Race: White, Age: 0-19: Sex: None, Population: Total}}
+    # {"C27001A_OO3E": {Race: White, Age: 0-19: Sex: None, PopulW[MetadataKey.RACE]
+    def build_metadata_list(self):
+        for race_prefix in HEALTH_INSURANCE_BY_RACE_GROUP_PREFIXES:
+            for prefix in race_prefix:
+                for suffix in HEALTH_INSURANCE_BY_RACE_GROUP_SUFFIXES:
+                    key = prefix+suffix
+                    prefix_metadata = race_prefix[prefix].copy()
+                    prefix_metadata.update(
+                        HEALTH_INSURANCE_BY_RACE_GROUP_SUFFIXES[suffix])
+                    self.metadata[key] = prefix_metadata
+
+        for sex_prefix in HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX:
+            for male_suffix in HEALTH_INSURANCE_BY_SEX_MALE_SUFFIXES:
+                key = sex_prefix + male_suffix
+                meta = HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX[sex_prefix].copy()
+                meta.update(HEALTH_INSURANCE_BY_SEX_MALE_SUFFIXES[male_suffix])
+                self.metadata[key] = meta
+
+            for female_suffix in HEALTH_INSURANCE_BY_SEX_FEMALE_SUFFIXES:
+                key = sex_prefix + female_suffix
+                meta = HEALTH_INSURANCE_BY_SEX_GROUPS_PREFIX[sex_prefix].copy()
+                meta.update(
+                    HEALTH_INSURANCE_BY_SEX_FEMALE_SUFFIXES[female_suffix])
+                self.metadata[key] = meta
+
+    # Pull the State_Fips map Code->Name from ACS
+
+    def get_state_fips_mapping(self):
+        params = {'for': 'state', "get": "NAME"}
+        resp = requests.get(self.base_url, params=params)
+        json_formatted_response = resp.json()
+        state_fips = {}
```

Comment:
I think you can get rid of the state_fips local variable and directly modify self.state_fips

ditto for county_fips in the method right below this one